src/node_util.cc

```diff
@@ -43,6 +44,22 @@ using v8::Value;
   VALUE_METHOD_MAP(V)
 #undef V
 
+static void GetPromiseDetails(const FunctionCallbackInfo<Value>& args) {
+  CHECK(args[0]->IsPromise());
+
+  auto isolate = args.GetIsolate();
+
+  Local<Promise> promise = args[0].As<Promise>();
+  Local<Array> ret = Array::New(isolate, 2);
```

Comment:
Besides the fact that this is simpler to implement, it is also the approach taken by the existing `GetProxyDetails`. Plus, I don't think it is very likely for promises to change its internal properties anytime soon.